AWSIoT/AWSIoTDataManager.m

```diff
@@ -332,6 +334,12 @@ - (instancetype)initWithConfiguration:(AWSServiceConfiguration *)configuration w
     return self;
 }
 
+- (nonnull NSString*)baseUserMetaDataString:(nullable NSString*)username {
+    return username.length
+    ? [NSString stringWithFormat:@"%@%@%@", username, @"?SDK=iOS&Version=", AWSIoTSDKVersion]
```

Comment:
You can simplify this by putting the hardcoded string as part of the format specifier: `[NSString stringWithFormat:@"%@?SDK=iOS&Version=%@", username, AWSIoTSDKVersion]`. You could simplify it further by assigning the username token then returning the same formatted string for each:

```objective-c
NSString *usernameComponent = username.length ? username : @"";
return [NSString stringWithFormat:@"%@?SDK=iOS&Version=%@", usernameComponent, AWSIoTSDKVersion];
```

My recollection is that interpolating a `nil` username into a string results in a string like `"(null)"` or similar, so the assignment is probably a good idea